### PR TITLE
feat: Migrate budget stack to CDK

### PR DESCRIPTION
**Pull request by gaudubey**
Migrate Budget Stack to CDK

### CODE REVIEW

File: cdk/test/functions/budget.test.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      templateFile: path.join(__dirname, '..', '..', '..', 'templates', 'budget.yaml'),
+    });
+  }
+}
+
+const context = {
+  [BUNDLING_STACKS]: [],
+};
+
+describe('resources', () => {
```
Comment (peermuellerxw) [inline cdk/test/functions/budget.test.ts:30]: I don't see where we actually compare the CFN resources with the ones synthezised from CDK. Such as https://github.com/superwerker/superwerker/blob/6e2af7291a22fe10617963d8321bc308d46c4461/cdk/test/guardduty.test.ts#L67

File: cdk/src/stacks/budget.ts [hunk 2]
```diff
@@ -1,13 +1,84 @@
...
+      policyDocument: budgetNotificationPolicyDocument,
+      topics: [budgetNotificationTopic],
+    });
+
+    const budgetAlarm = new cw.Alarm(this, 'Warning', {
+      comparisonOperator: cw.ComparisonOperator.GREATER_THAN_THRESHOLD,
+      alarmDescription: 'Superwerker default budget forecast exceed previous three months',
+      evaluationPeriods: 1,
+      threshold: 0,
+      treatMissingData: cw.TreatMissingData.MISSING,
```
Comment (peermuellerxw) [inline cdk/src/stacks/budget.ts:39]: maybe let's keep the original comment here also, do we know in the future why use this setting.

**Review by peermuellerxw — CHANGES_REQUESTED** [review]
Please also add an acceptance for this stack.

**Comment by aperov9** [comment]
main stack LGTM, just take a look at the unit & acceptance tests

**Review by peermuellerxw — APPROVED** [review]

**Pull request merged**